js/core/src/flow.ts

```diff
@@ -370,81 +333,263 @@ export class Flow<
+      } catch (e: any) {
+        const respBody = {
           error: {
-            status: 'INTERNAL',
-            message: getErrorMessage(e),
-            details: getErrorStack(e),
+            status: 'PERMISSION_DENIED',
+            message: e.message || 'Permission denied to resource',
           },
-        });
-        res.end();
+        };
+        response.status(403).send(respBody).end();
+        return;
       }
-    } else {
-      try {
-        const result = await this.invoke(input, { auth });
-        // Responses for non-streaming flows are passed back with the flow result stored in a field called "result."
-        res
-          .status(200)
-          .send({
-            result: result.result,
-          })
-          .end();
-      } catch (e) {
-        // Errors for non-streaming flows are passed back as standard API errors.
-        res
-          .status(500)
-          .send({
+
+      if (stream === 'true') {
+        response.writeHead(200, {
+          'Content-Type': 'text/plain',
+          'Transfer-Encoding': 'chunked',
+        });
+        try {
+          const result = await this.invoke(input, {
+            streamingCallback: ((chunk: z.infer<S>) => {
+              response.write(JSON.stringify(chunk) + streamDelimiter);
+            }) as S extends z.ZodVoid
+              ? undefined
+              : StreamingCallback<z.infer<S>>,
+            auth,
+          });
+          response.write({
+            result: result.result, // Need more results!!!!
+          });
+          response.end();
+        } catch (e) {
+          response.write({
             error: {
               status: 'INTERNAL',
               message: getErrorMessage(e),
               details: getErrorStack(e),
             },
-          })
-          .end();
+          });
+          response.end();
+        }
+      } else {
+        try {
+          const result = await this.invoke(input, { auth });
+          // Responses for non-streaming flows are passed back with the flow result stored in a field called "result."
+          response
+            .status(200)
+            .send({
+              result: result.result,
+            })
+            .end();
+        } catch (e) {
+          // Errors for non-streaming flows are passed back as standard API errors.
+          response
+            .status(500)
+            .send({
+              error: {
+                status: 'INTERNAL',
+                message: getErrorMessage(e),
+                details: getErrorStack(e),
+              },
+            })
+            .end();
+        }
       }
+    });
+  }
+}
+
+/**
+ * Options to configure the flow server.
+ */
+export interface FlowServerOptions {
+  /** Which environment(s) to run the flow server in. Defaults to `prod`. */
+  runInEnv?: 'all' | 'prod' | 'dev';
+  /** List of flows to expose via the flow server. If not specified, all registered flows will be exposed. */
+  flows?: Flow<any, any, any>[];
```

Comment:
this one is problematic... we can't have the flow ref when you're creating genkit instance... we need a different way to control which flows to expose in the flow server. maybe ignore for this PR.

Reply:
Ugh, I thought about this but then "realized" that since they're module-level, it has a fancier resolving mechanism but you're right. Even using string references won't help because it will still try to start up before they're defined. Bummer. Will think about options here.